sycl/source/detail/platform_impl.cpp

```diff
@@ -96,26 +96,28 @@ static bool IsBannedPlatform(platform Platform) {
          IsMatchingOpenCL(Platform, "AMD Accelerated Parallel Processing");
 }
 
-// This routine has the side effect of registering each platform's last device
-// id into each plugin, which is used for device counting.
-std::vector<platform> platform_impl::get_platforms() {
-
-  // Get the vector of platforms supported by a given UR plugin
-  // replace uses of this with with a helper in plugin object, the plugin
-  // objects will own the ur adapter handles and they'll need to pass them to
-  // urPlatformsGet - so urPlatformsGet will need to be wrapped with a helper
-  auto getPluginPlatforms = [](PluginPtr &Plugin) {
-    std::vector<platform> Platforms;
+// Get the vector of platforms supported by a given UR plugin
+// replace uses of this with a helper in plugin object, the plugin
+// objects will own the ur adapter handles and they'll need to pass them to
+// urPlatformsGet - so urPlatformsGet will need to be wrapped with a helper
+std::vector<platform> platform_impl::getPluginPlatforms(PluginPtr &Plugin,
+                                                        bool Supported) {
+  std::vector<platform> Platforms;
 
-    auto UrPlatforms = Plugin->getUrPlatforms();
+  auto UrPlatforms = Plugin->getUrPlatforms();
 
-    if (UrPlatforms.empty()) {
-      return Platforms;
-    }
+  if (UrPlatforms.empty()) {
+    return Platforms;
+  }
 
-    for (const auto &UrPlatform : UrPlatforms) {
-      platform Platform = detail::createSyclObjFromImpl<platform>(
-          getOrMakePlatformImpl(UrPlatform, Plugin));
+  for (const auto &UrPlatform : UrPlatforms) {
+    platform Platform = detail::createSyclObjFromImpl<platform>(
+        getOrMakePlatformImpl(UrPlatform, Plugin));
+    if (!Supported) {
+      if (IsBannedPlatform(Platform)) {
+        Platforms.push_back(Platform);
+      }
```

Comment:
Currently, we will print supported platforms with available devices, and unsupported/banned platforms with any number of devices. What about supported platforms with no devices?

Reply:
In all honesty I never though of it. It'll be easy enough to add.

Reply:
Done in: 03de71e8e0faeef59db9a38bf8e55b4cc0e7345e

Reply:
Thanks!